WordPress/Classes/ViewRelated/Media/MediaCellProgressView.swift

```diff
@@ -0,0 +1,111 @@
+import UIKit
+import WordPressShared
+
+/// Progress view displayed in cells in the media library to indicate that an
+/// asset is currently being processed, uploaded, or has failed to upload.
+///
+class MediaCellProgressView: UIView {
+    fileprivate let progressIndicator = ProgressIndicatorView()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        addProgressIndicator()
+
+        backgroundColor = WPStyleGuide.darkGrey()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func addProgressIndicator() {
+        addSubview(progressIndicator)
+        progressIndicator.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            progressIndicator.centerXAnchor.constraint(equalTo: centerXAnchor),
+            progressIndicator.centerYAnchor.constraint(equalTo: centerYAnchor)
+            ])
+    }
+}
+
+/// Small circular progress indicator view, currently just used in media cells
+/// during the processing / upload flow.
+///
+private class ProgressIndicatorView: UIView {
+    private let progressLayer = CAShapeLayer()
+
+    private enum Appearance {
```

Comment:
A struct looks feels more appropriate for constants than an enum, thoughts?

Reply:
Going by the RW style guide, the advantage of a caseless enum is that it can’t be accidentally instantiated: https://github.com/raywenderlich/swift-style-guide/blob/master/README.markdown#constants